auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -1294,6 +1414,206 @@ private void consumeContent(final XMLStreamReader reader) throws XMLStreamExcept
+                if (this.publicKey.getAlgorithm().equals("DSA")) {
+                    signatureMethod = ALGO_ID_SIGNATURE_DSA_SHA256;
+                } else if (this.publicKey.getAlgorithm().equals("RSA")) {
+                    signatureMethod = ALGO_ID_SIGNATURE_RSA_SHA256;
+                } else if (this.publicKey.getAlgorithm().equals("HMAC")) {
+                    signatureMethod = ALGO_ID_MAC_HMAC_SHA256;
+                } else if (this.publicKey.getAlgorithm().equals("EC")) {
+                    signatureMethod = ALGO_ID_SIGNATURE_ECDSA_SHA256;
+                }
+                SignedInfo si = fac.newSignedInfo
+                        (fac.newCanonicalizationMethod
+                                        (CanonicalizationMethod.INCLUSIVE,
+                                                (C14NMethodParameterSpec) null),
+                                fac.newSignatureMethod(signatureMethod, null),
+                                Collections.singletonList(ref));
+                KeyInfoFactory kif = fac.getKeyInfoFactory();
+                KeyValue kv = kif.newKeyValue(this.publicKey);
+                KeyInfo ki = kif.newKeyInfo(Collections.singletonList(kv));
+                XMLSignature signature = fac.newXMLSignature(si, ki);
+                signature.sign(dsc);
+                TransformerFactory transformerFactory = TransformerFactory.newInstance();
+                Transformer transformer = transformerFactory.newTransformer();
+                DOMSource source = new DOMSource(doc);
+                FileWriter writer = new FileWriter(path);
+                StreamResult result = new StreamResult(writer);
+                transformer.transform(source, result);
+                ElytronMessages.log.tracef("FileSystemSecurityRealm - signature against file updated [%s]", name);
+                writer.close();
+
+            } catch (ParserConfigurationException | IOException | NoSuchAlgorithmException | InvalidAlgorithmParameterException |
+                    KeyException | XMLSignatureException | MarshalException | TransformerException | SAXException e) {
+                throw ElytronMessages.log.unableToGenerateSignature(path);
+            }
+        }
+
+        private static void writeToMainIndex(File mainIndex, Path path, SecretKey secretKey) throws IOException, GeneralSecurityException {
+            FileWriter mainIndexWriter = new FileWriter(mainIndex, true);
+            MessageDigest messageDigest = MessageDigest.getInstance("MD5");
+            String checksum = getChecksum(messageDigest, new File(path.toString()));
+            String identityFilename = path.getFileName().toString();
+            if(secretKey != null) {
+                checksum = CipherUtil.encrypt(checksum, secretKey);
+                identityFilename = CipherUtil.encrypt(identityFilename, secretKey);
+            }
+            mainIndexWriter.append(identityFilename).append(":").append(checksum).append('\n');
```

Comment:
`System.lineSeparator()` should be used instead of `\n` so this works properly on Windows too.